extensions/sftp/CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+find_package(CURL REQUIRED)
+
+include(${CMAKE_SOURCE_DIR}/extensions/ExtensionHeader.txt)
+include_directories(client processors)
+
+file(GLOB SOURCES  "*.cpp" "client/*.cpp" "processors/*.cpp")
+
+if (USE_CURL_NSS)
+    message("Using NSS")
+	add_definitions(-DUSE_CURL_NSS)
+endif()
+
+add_library(minifi-sftp STATIC ${SOURCES})
+set_property(TARGET minifi-sftp PROPERTY POSITION_INDEPENDENT_CODE ON)
+if(THREADS_HAVE_PTHREAD_ARG)
+	target_compile_options(PUBLIC minifi-sftp "-pthread")
+endif()
+if(CMAKE_THREAD_LIBS_INIT)
+	target_link_libraries(minifi-sftp "${CMAKE_THREAD_LIBS_INIT}")
+endif()
+
+if (CURL_FOUND)
+	include_directories(${CURL_INCLUDE_DIRS})
+	if (WIN32)
+		message("Including ${CURL_LIBRARY}")
+		target_link_libraries (minifi-sftp ${CURL_LIBRARY})
+		set_target_properties(minifi-sftp PROPERTIES LINK_FLAGS "${LINK_FLAGS} /WHOLEARCHIVE:${CURL_LIBRARY}")
+	else()
+		target_link_libraries(minifi-sftp ${CURL_LIBRARIES})
+	endif()
+endif ()
+
+# Include UUID
+find_package(UUID REQUIRED)
```

Comment:
A lot of extensions were copy/paste, but i don't think things like uuid are required everywhere since these aren't (currently) releasable targets. In any case you can leave it here unless you have a burning desire to remove them.

Reply:
Could you elaborate on this please? PutSFTP's contructor has a (default) utils::Identifier uuid parameter which with the delegated Processor constructor is called. Is this unnecessary? What do you mean by uuid only needed in releasable targets?

Reply:
Sorry that wasn't clear, UUID is transitively included via libminifi -- and at this point the extensions are not releasable -- but I plan on them being so at some point. We can discuss this further in tickets. I don't think a change is needed it was more FYI